addons/account_debit_note/wizard/account_debit_note.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+from odoo import models, fields, api
+from odoo.tools.translate import _
+
+
+class AccountDebitNote(models.TransientModel):
+    """
+    Add Debit Note wizard: when you want to correct an invoice with a tiny amount of something
+    """
+    _name = 'account.debit.note'
+    _description = 'Add Debit Note wizard'
+
+    move_id = fields.Many2one('account.move', string='Journal Entry',
+        domain=[('state', '=', 'posted'), ('type', 'not in', ('out_refund', 'in_refund'))])
+    date = fields.Date(string='Adjustment date', default=fields.Date.context_today, required=True)
+    reason = fields.Char(string='Reason')
+    journal_id = fields.Many2one('account.journal', string='Use Specific Journal', help='If empty, uses the journal of the journal entry to be reversed.')
+
+    # computed fields
+    residual = fields.Monetary(compute="_compute_from_moves")
+    currency_id = fields.Many2one('res.currency', compute="_compute_from_moves")
+    move_type = fields.Char(compute="_compute_from_moves")
+
+    @api.model
+    def default_get(self, fields):
+        res = super(AccountDebitNote, self).default_get(fields)
+        move_ids = self.env['account.move'].browse(self.env.context['active_ids']) if self.env.context.get('active_model') == 'account.move' else self.env['account.move']
+        res['residual'] = len(move_ids) == 1 and move_ids.amount_residual or 0
+        res['currency_id'] = len(move_ids.currency_id) == 1 and move_ids.currency_id.id or False
+        res['move_type'] = len(move_ids) == 1 and move_ids.type or False
+        return res
+
+    @api.depends('move_id')
+    def _compute_from_moves(self):
+        move_ids = self.env['account.move'].browse(self.env.context['active_ids']) if self.env.context.get('active_model') == 'account.move' else self.move_id
+        for record in self:
+            record.residual = len(move_ids) == 1 and move_ids.amount_residual or 0
+            record.currency_id = len(move_ids.currency_id) == 1 and move_ids.currency_id or False
+            record.move_type = len(move_ids) == 1 and move_ids.type or False
+
+    def _prepare_default_values(self, move):
+        return {
+                'ref': _('Adjustment of: %s, %s') % (move.name, self.reason) if self.reason else _('Adjustment of: %s') % (move.name),
```

Comment:
It can sound strong, but this string should not be wired... A debit note is not an **Adjustment** by default, the reason itself is enough, no concatenation here is necessary I think.